webdriver/tests/bidi/script/call_function/arguments.py

```diff
@@ -0,0 +1,116 @@
+        ({"type": "number", "value": 1.4}, "1.4"),
+        ({"type": "boolean", "value": True}, "true"),
+        ({"type": "boolean", "value": False}, "false"),
+        ({"type": "bigint", "value": "42"}, "42n"),
+    ],
+)
+async def test_primitive_values(bidi_session, top_context, argument, expected):
+    result = await bidi_session.script.call_function(
+        function_declaration=
+        f"(arg) => {{ "
+        f"    if(arg!=={expected}) "
+        f"        throw Error(\"Argument type should be {expected}, but was \"+arg);"
+        f"    return arg;"
+        f"}}",
+        arguments=[argument],
+        await_promise=False,
+        target=ContextTarget(top_context["context"]),
+    )
+
+    assert result == argument
+
+
+@pytest.mark.asyncio
+async def test_nan(bidi_session, top_context):
+    nan_remote_value = {"type": "number", "value": "NaN"}
+    result = await bidi_session.script.call_function(
+        function_declaration=
+        f"(arg) => {{ "
+        f"    if(!isNaN(arg)) "
+        f"        throw Error(\"Argument type should be 'NaN', but was \"+arg);"
+        f"    return arg;"
+        f"}}",
+        arguments=[nan_remote_value],
+        await_promise=False,
+        target=ContextTarget(top_context["context"]),
+    )
+
+    assert result == nan_remote_value
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "argument, expected_type",
+    [
+        ({
+             "type": "array",
+             "value": [
+                 {"type": "string", "value": "foobar"},
+             ],
+         },
+         "Array"
+        ),
+        ({"type": "date", "value": "2022-05-31T13:47:29.000Z"},
+         "Date"
+         ),
+        ({
+             "type": "map",
+             "value": [
+                 ["foobar", {"type": "string", "value": "foobar"}],
+             ],
+         },
+         "Map"
+        ),
+        ({
+             "type": "object",
+             "value": [
+                 ["foobar", {"type": "string", "value": "foobar"}],
+             ],
+         },
+         "Object"
+        ),
+        ({"type": "regexp", "value": {"pattern": "foo", "flags": "g"}},
+         "RegExp"
+         ),
+        ({
+             "type": "set",
+             "value": [
+                 {"type": "string", "value": "foobar"},
+             ],
+         },
+         "Set"
+        )
+    ],
+)
+async def test_local_values(bidi_session, top_context, argument, expected_type):
+    result = await bidi_session.script.call_function(
+        function_declaration=
+        f"(arg) => {{ "
+        f"    if(! (arg instanceof {expected_type})) "
+        f"        throw Error(\"Argument type should be {expected_type}, but was \"+"
+        f"            Object.prototype.toString.call(arg));"
+        f"    return arg;"
+        f"}}",
+        arguments=[argument],
+        await_promise=False,
+        target=ContextTarget(top_context["context"]),
+    )
+
+    assert result == argument
```

Comment:
I think we should rather check type & value here. The spec does not guarantee that a LocalValue and a RemoteValue can be compared safely.

Reply:
Switched to `recursive_compare`.